### PR TITLE
Subtask coordinate reward rule update & continual training

### PR DESCRIPTION
**Pull request by chc012**
Updates
- Update all training yamls (7B, 32B, 72B) for subtasks.
- Update coordinate reward rules to specify Gaussian sigma; add pixel square metric (whether the predicted coordinate is within a square around the ground truth pixel, to mimic a small bounding box).
- Update unit tests for the reward rules
- other minor fixes

Tests
- Trained 32B, 7B, and 72B models with results in https://docs.google.com/spreadsheets/d/1rk8VSM_7lytOBjAOiSrNaxugfmd0Qk3GYHPCalIemaQ/edit?usp=sharing

### CODE REVIEW

File: orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  cd /workspace/verl
+  sed -i 's|mirrors.tuna.tsinghua.edu.cn|us.archive.ubuntu.com|g' /etc/apt/sources.list
+  apt update
+  apt install iproute2 -y
+  apt install -y dnsutils
+  apt install -y emacs
+  apt install -y awscli
+  pip install 'urllib3<2'
+  pip install s3fs
+  pip install parquet-tools
```
Comment (manpk-ai) [inline orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml:42]: Can we add installation of these packages to requirements.txt?

File: orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  export REWARD_FN="SET THE REWARD FUNCTION NAME HERE"
+  export REWARD_FILE=orby/reward/subtask.py
+
+  cd /workspace/verl
+  sed -i 's|mirrors.tuna.tsinghua.edu.cn|us.archive.ubuntu.com|g' /etc/apt/sources.list
+  apt update
+  apt install iproute2 -y
+  apt install -y dnsutils
+  apt install -y emacs
+  apt install -y awscli
```
Comment (manpk-ai) [inline orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml:39]: Some of these packages are already being installed in the init_interactive.sh

File: orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml (hunk 1 above)
Comment (chc012) [reply orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml:42]: We actually never uses the requirement.txt. We install verl directly in the line
```
# Install verl lib: https://verl.readthedocs.io/en/latest/start/install.html
pip3 install -e .[vllm]
```
Also, Some of these packages (boto3, s3fs) actually gives you bug when you install it along with other things in the requirement.txt.

File: orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml (hunk 2 above)
Comment (chc012) [reply orby/mcli/ray_train_qwen_2_5_vl_72B_subtask.yaml:39]: We don't run `init_interactive.sh` here. That bash script is used specifically for interactive sessions. Here we are running a training mosaic session so we have to install everything just like we did manually in an interactive session by running `init_interactive.sh`.

File: orby/mcli/ray_train_qwen_2_5_vl_7B_subtask.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            data.truncation='error' \
+            data.image_key=images \
+            actor_rollout_ref.model.path=$MODEL_NAME \
+            actor_rollout_ref.actor.optim.lr=1e-6 \
+            actor_rollout_ref.model.use_remove_padding=True \
+            actor_rollout_ref.actor.ppo_mini_batch_size=$TRAIN_BATCH_SIZE \
+            actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=4 \
+            actor_rollout_ref.actor.ppo_max_token_len_per_gpu=8192 \
+            actor_rollout_ref.actor.use_kl_loss=True \
+            actor_rollout_ref.actor.kl_loss_coef=0.01 \
```
Comment (manpk-ai) [inline orby/mcli/ray_train_qwen_2_5_vl_7B_subtask.yaml:93]: Is 0.01 a standard hyperparam value for kl_loss_coeff?

Comment (chc012) [reply orby/mcli/ray_train_qwen_2_5_vl_7B_subtask.yaml:93]: This is adopted from @sanjari-orb's general training yaml. We haven't tune these things yet (which we probably should at some point cc @ligang-orby). But should work as a starting point.

File: orby/reward/subtask.py [hunk 4]
```diff
@@ -349,30 +435,134 @@ def score(
...
+            ground_truth,
+            detailed=False,
+            coordinates_metric="gaussian",
+            coordinates_gaussian_sigma=5,
+        )
+    else:
+        raise NotImplementedError
+
+
+def training_reward_func_gaussian_sigma_2(
```
Comment (ligang-orby) [inline orby/reward/subtask.py:502]: One trick I'm validating for passing extra args to the reward fn (both training and eval): https://github.com/orby-ai-engineering/verl/blob/dev/orby/scripts/eval_screenspot.sh#L141
https://github.com/orby-ai-engineering/verl/blob/dev/orby/reward/screenspot.py#L295

Comment (chc012) [reply orby/reward/subtask.py:502]: Oh nice. Does this work for training yaml too?

Comment (ligang-orby) [reply orby/reward/subtask.py]: Testing in thie PR: https://github.com/orby-ai-engineering/verl/pull/24 -- should be done by today. Will update once all verification is done

**Comment by chc012** [comment]
waiting for https://github.com/orby-ai-engineering/verl/pull/24

**Comment by ligang-orby** [comment]
https://github.com/orby-ai-engineering/verl/pull/24 is verified and submitted
